server/socketio-server.js

```diff
@@ -83,7 +96,29 @@ socketAuth(io, {
     io.emit('welcome', {
       message: `Hello, your socket ID is ${socket.id}`,
     });
+
+    socket.on('send-message', async (data) => {
+      // save messages to mongo -> in users conversation
+      const newMessage = await updateConversation(
+        data.senderId,
+        data.receiverId,
+        data.message,
+      );
+      const sockets = getReceiverSockets(data.receiverId);
+      console.log('SOCKETS', sockets);
+      if (!sockets) {
+        // user offline - notification
+        console.debug('User offline. Message saved.');
+      } else {
+        // user online - no notification
+        sockets.forEach((socketId) => {
+          console.debug('Emitting message to receiver');
+          socket.to(socketId).emit('receive-message', newMessage);
+        });
+      }
+    });
```

Comment:
I don't really know if it matters or makes a difference but I was able to get my socket working outside of the 'postAuthenticate' method in this socketAuth function. In the grand scheme of things it doesn't really matter, I guess if we wanted to keep things that don't deal with authentication separate from the socketAuth function we could put it outside of it which worked for me [here](https://github.com/hatchways/team-lucky-charms/pull/54/files#r532951261).

I know we were struggling with it initially not working outside of the postAuthenticate method so like I said not super important but I figured I'd let you know.

Reply:
True, it would definitely be much more organized to keep postAuth method just for authentication. But i was wondering since messaging should be only for authenticated users (sockets), it should be restricted to postAuth (only authenticated users). However, I put the function body outside just to clean up postAuth.

Reply:
That's a really good point! I hadn't thought of that. I'm good with keeping it there in that case.